onc-healthit/lantern-back-end
# Supported resources

Changes in this PR:

A supported resources field has been added to the endpoint info struct which is then added to the fhir_endpoints_info table and fhir_endpoints_info_history table in the db. This field is an array that stores all the resource types that a given capability statement supports. Testing also was added/updated to support these changes. 

TO TEST:
Start up the lantern project normally with make run and make sure the db is populated. Wait for capability reciever to start adding data into the fhir_endpoints _info table. All entries that include a capability statement (not null) should also have a supported fields array (not null) and this array should contain all the resource types of this capability statement

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: 
[Lantern 215 JIRA ticket](https://oncprojectracking.healthit.gov/support/browse/LANTERN-215)
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
